### PR TITLE
Entity search

### PR DESCRIPTION
**Pull request by rakeshSgr**

### CODE REVIEW

File: src/controllers/v1/entity.js [hunk 1]
```diff
@@ -81,4 +81,22 @@ module.exports = class Entity {
...
+
+	/**
+	 * entity details
+	 * @method
+	 * @name details
+	 * @param {Object} req - request data.
+	 * @returns {JSON} - entities.
+	 */
+
+	async details(req) {
```
Comment (aks30) [inline src/controllers/v1/entity.js:93]: @rakeshSgr Let's rename it to list

File: src/controllers/v1/entity.js [hunk 2]
```diff
@@ -81,4 +81,22 @@ module.exports = class Entity {
...
+		try {
+			req.decodedToken = {}
+			req.decodedToken.id = 0
```
Comment (aks30) [inline src/controllers/v1/entity.js:96]: Not sure what is this about?

File: src/database/queries/entity.js [hunk 3]
```diff
@@ -55,4 +57,21 @@ module.exports = class UserEntityData {
 			return error
 		}
 	}
+
+	static async getAllEntities(filters, attributes, page, limit, search) {
```
Comment (aks30) [inline src/database/queries/entity.js:61]: Can we extend and reuse the function - findAllEntities

File: src/services/entity.js [hunk 4]
```diff
@@ -204,4 +204,65 @@ module.exports = class EntityHelper {
...
+	 * @param {String} searchText - search label in entity.
+	 * @param {Integer} page -  page no.
+	 * @param {Integer} pageSize -  page limit per api.
+	 * @returns {JSON} - Entity search matched response.
+	 */
+	static async details(query, userId, searchText, pageNo, pageSize) {
+		try {
+			let entityType = query.entity_type_id ? query.entity_type_id : ''
+			let filter
+			if (query.read_user_entity == true) {
```
Comment (aks30) [inline src/services/entity.js:223]: What does the key read_user_entity mean? When will it be true and false?

**Review by aks30 — CHANGES_REQUESTED** [review]
Reviewed on 1 July

File: src/services/entity.js (hunk 4 above)
Comment (rakeshSgr) [reply src/services/entity.js:223]: read_user_entity = = true, it reads requested user created entity and entity created by id 0, 
which means default entities,
The same logic is there in existing read api also

File: src/database/queries/entity.js (hunk 3 above)
Comment (rakeshSgr) [reply src/database/queries/entity.js:61]: that returns only data,  we need data with count as well.  we can do that

Comment (rakeshSgr) [reply src/database/queries/entity.js:61]: we have pagination so it won't return all data,  we can't use it

File: src/controllers/v1/entity.js (hunk 1 above)
Comment (rakeshSgr) [reply src/controllers/v1/entity.js]: done

File: src/controllers/v1/entity.js (hunk 2 above)
Comment (rakeshSgr) [reply src/controllers/v1/entity.js]: resolved

File: src/services/entity.js [hunk 5]
```diff
@@ -204,4 +204,65 @@ module.exports = class EntityHelper {
...
+				filter = {
+					[Op.or]: [
+						{
+							created_by: '0',
+						},
+						{
+							created_by: userId,
```
Comment (aks30) [inline src/services/entity.js:230]: Will there be ever a case when regular user will have put an entry in entity table that we have to filter by his userId?

**Review by aks30 — CHANGES_REQUESTED** [review]
Reviewed on 2 July

File: src/services/entity.js (hunk 5 above)
Comment (rakeshSgr) [reply src/services/entity.js]: currently its not require,  in the beginning  user can also create entities in that this filter was added

Comment (rakeshSgr) [reply src/services/entity.js]: frontend not using this

File: src/controllers/v1/entity.js [hunk 6]
```diff
@@ -81,4 +81,20 @@ module.exports = class Entity {
...
+	 * entity list
+	 * @method
+	 * @name list
+	 * @param {Object} req - request data.
+	 * @returns {JSON} - entities.
+	 */
+
+	async list(req) {
+		try {
+			return await entityService.list(req.query, req.decodedToken.id, req.searchText, req.pageNo, req.pageSize)
```
[the remark this answers is not in the document]
Comment (rakeshSgr) [reply src/controllers/v1/entity.js]: removed this, not require

**Pull request merged**